programs/spl-token-staking/src/instructions/update_token_meta.rs

```diff
@@ -0,0 +1,91 @@
+use crate::{errors::ErrorCode, stake_pool_signer_seeds, state::StakePool, ID};
+use anchor_lang::prelude::*;
+use anchor_spl::metadata::{
+    create_metadata_accounts_v3, update_metadata_accounts_v2, CreateMetadataAccountsV3,
+    UpdateMetadataAccountsV2,
+};
+use mpl_token_metadata::state::DataV2;
+
+#[derive(Accounts)]
+pub struct UpdateTokenMeta<'info> {
+    pub payer: Signer<'info>,
+
+    /// CHECK: Handled by metadata program
+    #[account(mut)]
+    pub metadata_account: UncheckedAccount<'info>,
+
+    #[account(
+      owner = ID,
+      // Validates the StakePool's stake mint matches the mint to have updated metadata
+      has_one = stake_mint
+    )]
+    pub stake_pool: AccountLoader<'info, StakePool>,
+
+    /// CHECK: Handled by has_one with stake_mint
+    pub stake_mint: UncheckedAccount<'info>,
+
+    /// CHECK: Handled by address check
+    #[account(
+      address = mpl_token_metadata::ID
+    )]
+    pub metadata_program: UncheckedAccount<'info>,
+    pub rent: Sysvar<'info, Rent>,
+    pub system_program: Program<'info, System>,
+}
+
+pub fn handler(
+    ctx: Context<UpdateTokenMeta>,
+    name: String,
+    symbol: String,
+    uri: String,
+) -> Result<()> {
+    let stake_pool = ctx.accounts.stake_pool.load()?;
+
+    require!(
+        ctx.accounts.payer.key() == stake_pool.authority,
+        ErrorCode::InvalidAuthority
+    );
```

Comment:
why not have this in a constraint in the accounts above?

Reply:
copy pasta from elsewhere. Will update